pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -456,6 +456,14 @@ func (b *BridgeBindMechanism) startDHCP() error {
 }
 
 func (b *BridgeBindMechanism) preparePodNetworkInterfaces() error {
+	if b.wasCachedInterfaceLoaded() {
```

Comment:
I don't understand what functionality this commit changes.
It looks for me exactly the same as it was before. What do I miss?

IIUC, the bug that we have can happen if virt-handler crashes during `preparePodNetworkInterfaces` but before `bindMechanism.setCachedInterface()`. Then on the re-sync, in the good case we will get an error, in the bad case we will end up with an undesirable virt-launcher network configuration. For example if the `preparePodNetworkInterfaces` fails after bridge binding created a new `dummy` interface and gave it the name of `eth0`.

Reply:
I don't understand your concept of good or bad. 

Setting the `virtIface` attribute of the bind mechanism is the last thing done in this method (and only when the method finishes successfully we do set its `Target` attribute); as such, if you happen to re-enter here, it can be based on 2 things:
  - prepare network interfaces did **not** finish successfully: it will have the `virtIface` attribute initialized, but it will not feature the target attribute within the virtIface. As such, we throw a the critical network error, meaning there won't be another re-try.
  - prepare network interfaces **did** finish successfully: it will feature the `Target` attribute in the `virtIface`. As such, we just return early, since the networking infrastucture has been already created.

Is is clearer now ? I agree that the commit msg of the 1st commit is misleading, I will update it.

Reply:
Good - fixing the mentioned bug.
Bad - complicating the code without fixing any issue.

`preparePodNetworkInterfaces` may be re-executed only if virt-launcher crashed during the sync.
In this case we have 2 options -
1. `virtIface` is cached - all good, it means the `prepare` finished successfully the last time, we can skip `PlugPhase1`.
2. `virtIface` is not cached - in this case we don't know if the `Prepare` did changes on the virt-launcher network pod the last time it was/wasn't executed. So we may have a bug in this flow.
On re-sync, a new instance of `virtIface` is created, so I truly don't understand how setting or not setting the target on the old non cached instance may help.

Reply:
You're right, I somehow forgot that caching must be manually called.

I'll remove the first commit from this PR, and we can address it at a later stage.